api/lib/infrastructure/utils/csv/certification-results.js

```diff
@@ -0,0 +1,101 @@
+const { getCsvContent } = require('./write-csv-utils');
+const { status: assessmentResultStatuses } = require('../../../domain/models/AssessmentResult');
+const _ = require('lodash');
+const moment = require('moment');
+
+const competenceIndexes = [
+  '1.1', '1.2', '1.3',
+  '2.1', '2.2', '2.3', '2.4',
+  '3.1', '3.2', '3.3', '3.4',
+  '4.1', '4.2', '4.3',
+  '5.1', '5.2',
+];
+
+function _sortCompetenceMarkByCompetenceIndex({ competencesWithMark }) {
+  return competencesWithMark.reduce((result, competence) => {
+    const competenceCode = competence.competence_code;
+    result[competenceCode] = { level: competence.level };
+    return result;
+  }, {});
+}
+
+function _getCompetenceLevel({ competencesWithMark, competenceIndex, isCertifRejected }) {
+  const certificationIndexedCompetences = _sortCompetenceMarkByCompetenceIndex({ competencesWithMark });
+  const competence = certificationIndexedCompetences[competenceIndex];
+
+  const _competenceIsFailedOrCertifRejected =
+    (competence) => competence.level === 0 || competence.level === -1 || isCertifRejected;
```

Comment:
mini optimisation (on commence par le boolean qui est plus rapide, si vrai, alors on ne verifie pas la valeur de level): 

```suggestion
  const _competenceIsFailedOrCertifRejected = ({level}) => isCertifRejected || level < 1;
```